src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineView.xaml.cs

```diff
@@ -0,0 +1,199 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.VisualStudio.Editor;
+using Microsoft.VisualStudio.LanguageServices.Implementation;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.TextManager.Interop;
+
+namespace Microsoft.VisualStudio.LanguageServices.DocumentOutline
+{
+    /// <summary>
+    /// Interaction logic for DocumentOutlineView.xaml
+    /// All operations happen on the UI thread for visual studio
+    /// </summary>
+    internal sealed partial class DocumentOutlineView : UserControl, IVsCodeWindowEvents, IDisposable
+    {
+        private readonly IVsCodeWindow _codeWindow;
+        private readonly IThreadingContext _threadingContext;
+        private readonly DocumentOutlineViewModel _viewModel;
+        private readonly IVsEditorAdaptersFactoryService _editorAdaptersFactoryService;
+        private readonly Dictionary<IVsTextView, ITextView> _trackedTextViews = new();
```

Comment:
@jasonmalinowski if we can extract out this special split view window handling into a helper type that can be used in these locations.